src/mock.ts

```diff
@@ -91,57 +101,68 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
   // and completes the customMock object with any fields
   // defined on genericMock
   // only merges objects or arrays. Scalars are returned as is
-  function mergeMocks(genericMockFunction, customMock) {
+  function mergeMocks(genericMockFunction: IMockFn, customMock: any): any {
     if (Array.isArray(customMock)) {
-      return customMock.map((el) => mergeMocks(genericMockFunction, el));
+      return customMock.map((el: any) => mergeMocks(genericMockFunction, el));
     }
     if (isObject(customMock)) {
       return mergeObjects(genericMockFunction(), customMock);
     }
     return customMock;
   }
 
-  function assignResolveType(type) {
+  function getResolveType(namedFieldType: GraphQLNamedType) {
+    if ( (namedFieldType instanceof GraphQLInterfaceType) ||
+         (namedFieldType instanceof GraphQLUnionType) ) {
+        return namedFieldType.resolveType;
+    } else {
+        return undefined;
+    }
+  }
+
+  function assignResolveType(type: GraphQLType) {
     const fieldType = getNullableType(type);
     const namedFieldType = getNamedType(fieldType);
 
-    const oldResolveType = namedFieldType.resolveType;
+    const oldResolveType = getResolveType(namedFieldType);
     if (preserveResolvers && oldResolveType && oldResolveType.length) {
       return;
     }
 
     if (namedFieldType instanceof GraphQLUnionType ||
-        namedFieldType instanceof GraphQLInterfaceType
-    ) {
+        namedFieldType instanceof GraphQLInterfaceType) {
       // the default `resolveType` always returns null. We add a fallback
       // resolution that works with how unions and interface are mocked
-      namedFieldType.resolveType = (data, context, info) =>
-        info.schema.getType(data.typename);
+      namedFieldType.resolveType = (data: any, context: any, info?: GraphQLResolveInfo) => {
+        if ( undefined === info ) {
```

Comment:
Ok,
so doing extra check, it looks like the '?' comes from graphql's typings.
giving a quick glance at graphql-js's code, i couldn't found a flow which is undefined,
what do you suggest to do? keep it or remove it from the typings (and the code)..